File: C#/2024/2024-002/main.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;

class SafeReports
{
    // Method to check if a report is safe
    static bool IsSafe(List<int> report)
    {
        var differences = report.Zip(report.Skip(1), (a, b) => b - a).ToList();
        
        bool allIncreasing = differences.All(diff => diff >= 1 && diff <= 3);
        bool allDecreasing = differences.All(diff => diff <= -1 && diff >= -3);

        return allIncreasing || allDecreasing;
    }

    // Method to check if a report is safe with the Problem Dampener
    static bool IsSafeWithDampener(List<int> report)
    {
        if (IsSafe(report))
        {
            return true;
        }

        for (int i = 0; i < report.Count; i++)
        {
            var modifiedReport = report.Take(i).Concat(report.Skip(i + 1)).ToList();
            if (IsSafe(modifiedReport))
            {
                return true;
            }
        }

        return false;
    }

    static void Main(string[] args)
    {
        string inputPath = "input.txt";

        // Read input lines from the file
        var lines = File.ReadAllLines(inputPath);
        var reports = lines.Select(line => line.Split(' ').Select(int.Parse).ToList()).ToList();

        // Count safe reports
        int safeCount = reports.Count(IsSafe);
        Console.WriteLine($"Safe reports: {safeCount}");

        // Count safe reports with the Problem Dampener
        int safeWithDampenerCount = reports.Count(IsSafeWithDampener);
        Console.WriteLine($"Safe reports with dampener: {safeWithDampenerCount}");
    }
}

File: PHP/2024/2024-002/main.php
<?php
// Function to check if a report is safe
function is_safe($report) {
    $differences = [];
    for ($i = 0; $i < count($report) - 1; $i++) {
        $differences[] = $report[$i + 1] - $report[$i];
    }
    
    $all_increasing = array_reduce($differences, function($carry, $diff) {
        return $carry && ($diff >= 1 && $diff <= 3);
    }, true);

    $all_decreasing = array_reduce($differences, function($carry, $diff) {
        return $carry && ($diff <= -1 && $diff >= -3);
    }, true);

    return $all_increasing || $all_decreasing;
}

// Function to check if a report is safe with the Problem Dampener
function is_safe_with_dampener($report) {
    if (is_safe($report)) {
        return true;
    }

    for ($i = 0; $i < count($report); $i++) {
        $modified_report = $report;
        array_splice($modified_report, $i, 1);
        if (is_safe($modified_report)) {
            return true;
        }
    }

    return false;
}

// Main function to process the input and count safe reports
function main() {
    $input_path = 'input.txt';

    // Read the input file
    $lines = file($input_path, FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    $reports = array_map(function($line) {
        return array_map('intval', preg_split('/\s+/', $line));
    }, $lines);

    // Count safe reports
    $safe_count = count(array_filter($reports, 'is_safe'));
    echo "Safe reports: $safe_count\n";

    // Count safe reports with the Problem Dampener
    $safe_with_dampener_count = count(array_filter($reports, 'is_safe_with_dampener'));
    echo "Safe reports with dampener: $safe_with_dampener_count\n";
}

// Run the main function
main();
